src/operator/race-static.ts

```diff
@@ -8,14 +8,14 @@ import {isArray} from '../util/isArray';
  * @param {...Observables} ...observables sources used to race for which Observable emits first.
  * @returns {Observable} an Observable that mirrors the output of the first Observable to emit an item.
  */
-export function race<T>(...observables: Array<Observable<any> | Array<Observable<any>>>): Observable<T> {
+export function race<T>(...observables: Array<Observable<T> | Array<Observable<T>>>): Observable<T> {
   // if the only argument is an array, it was most likely called with
   // `pair([obs1, obs2, ...])`
   if (observables.length === 1) {
-    if (isArray(observables[0])) {
-      observables = <Array<Observable<any>>>observables[0];
-    } else {
+    if (!isArray(observables[0])) {
```

Comment:
Ah, I thought this order (the following) would be more clearly because the previous one was 2->1->3 by the following items.

1. the case we don't need to create an `ArrayObservable` (return a result)
2. the case we need to override the `observables` argument.
3. create and return `ArrayObservable`.

If you don't favor in it, I'll revert this change.